src/Lecture4.hs

```diff
@@ -157,7 +202,13 @@ string.
 If both strings have the same length, return the first one.
 -}
 instance Semigroup MaxLen where
+   (<>) = getMaxLen
 
+getMaxLen :: MaxLen  -> MaxLen -> MaxLen
+getMaxLen m1 m2
+   | length (unMaxLen m1) > length (unMaxLen m2) = m1
+   | length (unMaxLen m1) < length (unMaxLen m2) = m2
+   | otherwise = m1
```

Comment:
This check can also be simplified a bit by using the `>=` operator
```suggestion
getMaxLen m1 m2
   | length (unMaxLen m1) >= length (unMaxLen m2) = m1
   | otherwise = m2
```